src/Type/UserAspectGetDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types = 1);
+
+namespace SaschaEgerer\PhpstanTypo3\Type;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\BooleanType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\IntegerRangeType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\StringType;
+use TYPO3\CMS\Core\Context\UserAspect;
+
+class UserAspectGetDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+
+	public function getClass(): string
+	{
+		return UserAspect::class;
+	}
+
+	public function isMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === 'get';
+	}
+
+	public function getTypeFromMethodCall(
+		MethodReflection $methodReflection,
+		MethodCall $methodCall,
+		Scope $scope
+	): ?\PHPStan\Type\Type
+	{
+		$firstArgument = $methodCall->args[0];
+
+		if (!$firstArgument instanceof Arg) {
+			return null;
+		}
+
+		$argumentType = $scope->getType($firstArgument->value);
+
+		if ($argumentType instanceof ConstantStringType) {
+			return match ($argumentType->getValue()) {
+				'id' => IntegerRangeType::createAllGreaterThanOrEqualTo(0),
+				'username' => new StringType(),
+				'isLoggedIn', 'isAdmin' => new BooleanType(),
+				'groupIds' => new ArrayType(new IntegerType(), IntegerRangeType::fromInterval(-2, null)),
+				'groupNames' => new ArrayType(new IntegerType(), new StringType()),
```

Comment:
I think we can solve this with a stub (or better annotations in the Core) instead using https://phpstan.org/blog/phpstan-1-6-0-with-conditional-return-types, which would require far less code.

Reply:
If you can tell me how this works, I am glad to implement it differently. AFAIK, you can only add one condition to implement a conditional return type like `@phpstan-return $raw is true ? array : QueryResultInterface`. Enlighten me please. :pray:

Reply:
That's a bit chaotic with 5 possible conditions, don't you think?

Reply:
That´s what i think.